azure/azure.go

```diff
@@ -284,11 +299,6 @@ func (a *azureOps) detachInternal(diskName, instance string) error {
 		newDataDisks = append(newDataDisks, d)
 	}
 
-	// If the disk is not attached, return success
```

Comment:
If we are getting rid of this check, then we need to add an equivalent check after updateDataDisks assuming in a normal scenario when (dataDisks == newDataDisks), the updateDataDisks API fails

Reply:
The API does not fail if you update the VM without any updates.